src/main/java/org/takes/rs/RsJSON.java

```diff
@@ -89,8 +90,13 @@ public RsJSON(final Response res) {
      */
     private static byte[] print(final RsJSON.Source src) throws IOException {
         final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        Json.createWriter(baos).write(src.toJSON());
-        return baos.toByteArray();
+        final JsonWriter writer = Json.createWriter(baos);
+        try {
+            writer.write(src.toJSON());
+            return baos.toByteArray();
```

Comment:
@dmzaytsev I think that `return` should go **after** `close()`. That's the whole point of this bug, isn't it? If you close the writer in `finally`, you will return an empty byte array.